<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>DeviceEditorDialog</class>
    <widget class="QDialog" name="DeviceEditorDialog">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>400</width>
                <height>300</height>
            </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
            <item>
                <widget class="QGroupBox" name="groupBox">
                    <property name="title">
                        <string>Device Details</string>
                    </property>
                    <layout class="QFormLayout" name="formLayout">
                        <item row="0" column="0">
                            <widget class="QLabel" name="label"><property name="text"><string>Name:</string></property></widget>
                        </item>
                        <item row="0" column="1">
                            <widget class="QLineEdit" name="lineEditName"/>
                        </item>

                        <item row="1" column="0">
                            <widget class="QLabel" name="label_2"><property name="text"><string>IP Address:</string></property></widget>
                        </item>
                        <item row="1" column="1">
                            <widget class="QLineEdit" name="lineEditIp"/>
                        </item>

                        <item row="2" column="0">
                            <widget class="QLabel" name="label_3"><property name="text"><string>Username:</string></property></widget>
                        </item>
                        <item row="2" column="1">
                            <widget class="QLineEdit" name="lineEditUser"/>
                        </item>

                        <item row="3" column="0">
                            <widget class="QLabel" name="label_4"><property name="text"><string>Platform:</string></property></widget>
                        </item>
                        <item row="3" column="1">
                            <widget class="QComboBox" name="cmbTarget">
                                <item>
                                    <property name="text"><string>NVIDIA TensorRT</string></property>
                                    <property name="userData"><string>nv-trt</string></property>
                                </item>
                                <item>
                                    <property name="text"><string>Rockchip RKNN</string></property>
                                    <property name="userData"><string>rockchip-rknn</string></property>
                                </item>
                                <item>
                                    <property name="text"><string>Intel OpenVINO</string></property>
                                    <property name="userData"><string>intel-ov</string></property>
                                </item>
                                <item>
                                    <property name="text"><string>AMD Vitis AI</string></property>
                                    <property name="userData"><string>amd-vitis</string></property>
                                </item>
                                <item>
                                    <property name="text"><string>Qualcomm QNN</string></property>
                                    <property name="userData"><string>qcom-qnn</string></property>
                                </item>
                            </widget>
                        </item>

                        <item row="4" column="0">
                            <widget class="QLabel" name="label_5"><property name="text"><string>Remote Path:</string></property></widget>
                        </item>
                        <item row="4" column="1">
                            <widget class="QLineEdit" name="lineEditPath">
                                <property name="text"><string>/home/user/xinfer_deploy/</string></property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <item>
                <widget class="QDialogButtonBox" name="buttonBox">
                    <property name="standardButtons">
                        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
                    </property>
                </widget>
            </item>
        </layout>
    </widget>
    <resources/>
    <connections>
        <connection>
            <sender>buttonBox</sender>
            <signal>accepted()</signal>
            <receiver>DeviceEditorDialog</receiver>
            <slot>on_btnSave_clicked()</slot>
        </connection>
        <connection>
            <sender>buttonBox</sender>
            <signal>rejected()</signal>
            <receiver>DeviceEditorDialog</receiver>
            <slot>on_btnCancel_clicked()</slot>
        </connection>
    </connections>
</ui>